app/models/course.rb

```diff
@@ -291,6 +291,85 @@ def watchlist_allow_ca
     watchlist_configuration.allow_ca
   end
 
+  def has_attachment?
+    !attachments.nil? && attachments.count > 0
+  end
+
+  def has_risk_conditions?
+    !risk_conditions.nil? && risk_conditions.count > 0
+  end
+
+  def has_watchlist_configuration?
+    watchlist_configuration != nil
+  end
+
+  def dump_yaml(include_metrics)
+    YAML.dump(serialize(include_metrics))
+  end
+
+  def generate_tar(export_configs)
+    base_path = Rails.root.join("courses", name).to_s
+    course_dir = name
+    attachments_dir = File.join(course_dir, "attachments")
+    rb_path = "course.rb"
+    config_path = "#{name}.yml"
+    mode = 0o755
+
+    begin
+      tarStream = StringIO.new("")
+      Gem::Package::TarWriter.new(tarStream) do |tar|
+        tar.mkdir course_dir, File.stat(base_path).mode
+
+        # save course.rb
+        source_file = File.open(File.join(base_path, rb_path), 'rb')
+        tar.add_file File.join(course_dir, rb_path), File.stat(source_file).mode do |tar_file|
+          tar_file.write(source_file.read)
+        end
+
+        # save course and metrics config
+        tar.add_file File.join(course_dir, config_path), mode do |tar_file|
+          tar_file.write(dump_yaml(export_configs&.include?('metrics_config')))
+        end
+
+        # save attachments
+        tar.mkdir attachments_dir, File.stat(base_path).mode
+        attachments.each do |attachment|
+          attachment_data = attachment.attachment_file.download
+          filename = attachment.filename
+          relative_path = File.join(attachments_dir, filename)
+
+          tar.add_file relative_path, mode do |file|
+            file.write(attachment_data)
+          end
+        end
+
+        # save assessments
+        if export_configs&.include?('assessments')
```

Comment:
assessment doesn't actually save handins, just the folder

Reply:
Going to leave it like this for now, will keep it as a stretch goal